src/op_mode/interfaces.py

```diff
@@ -296,6 +305,114 @@ def _get_counter_data(ifname: typing.Optional[str],
 
     return ret
 
+def _get_kernel_data(raw, ifname = None, detail = False):
+    if ifname:
+        # Check if the interface exists
+        if not interface_exists(ifname):
+            raise vyos.opmode.Error(f"{ifname} does not exist!")
```

Comment:
```suggestion
            raise vyos.opmode.IncorrectValue(f"{ifname} does not exist!")
```

`vyos.opmode.Error` is the base class, actual errors should be as specific as possible. There are two errors that may apply here: `UnconfiguredObject` and `IncorrectValue`. I'm leaning towards `IncorrectValue` because in this case interfaces may not exist in the _configuration_ but still exist in the system.

Reply:
Thanks, change made!